packages/components/src/Chips/InternalChipDismissible/hooks/useInternalChipDismissible.ts

```diff
@@ -22,13 +19,110 @@ export function useInternalChipDismissible({
     chip => !selected.includes(chip.value),
   );
 
+  return { chipOptions, sortedVisibleChipOptions, availableChipOptions };
+};
+
+// Helper function for keyboard navigation
+const getNavigationData = (wrapper: HTMLElement, target: HTMLElement) => {
+  // Find all focusable elements
+  const chipElements = Array.from(
+    wrapper.querySelectorAll('[role="option"]'),
+  ) as HTMLElement[];
```

Comment:
one of the downsides to querying the DOM, we lose inferred types and have to cast everything. don't love it.